### ISSUE TITLE
(aws-rds): better support for DBInstance read replica in different region to primary (typescript).

### ISSUE THREAD

**Issue by ocrlgv** [aws/aws-cdk#12838]
Add better feature to create a read replica of a DB instance in another region

### Use Case

It seems creating an RDS replica in another region from the primary is natively not supported (at least in typescript). Using the following fails due to no support for "sourceRegion" parameter for the primary and errors out that the primary instance is not found even though the source identifier is given which would indicate the primary is in another region (in our case ap-southeast-2, replica is to be in eu-central-1)

```ts
         const rdsPrimaryInstance = rds.DatabaseInstanceBase.fromDatabaseInstanceAttributes(this, "rdsPrimary", {
               instanceIdentifier: primaryIdentifier,
               instanceEndpointAddress: primaryEndPointAddress,
               port: 5432,
               securityGroups: [ ec2.SecurityGroup.fromSecurityGroupId(this, "primarySG", "sg-0xxxxxxxxxx") ]
          });



         const rdsReadReplica = new rds.DatabaseInstanceReadReplica(this, "rdsReadReplica", {
             sourceDatabaseInstance: rdsPrimaryInstance,
             vpc: vpc
             instanceType: dbInstSize
         });
```

I tried using custom resources with a basic create and delete function below, but the issue here is that it requires a custom provider due to the time it takes for RDS to create the instance (timed it to be over 18min) which would usually timeout Lambda. My stack depends on the returned values of endpoint, port, identifier, etc but aren't available as custom resource is asynchronous and return immediately.

```ts
                // Read Replica custom resource
                const rdsKms = kms.Alias.fromAliasName(this, 'rdsEncryptionKey', 'alias/aws/rds');
                // Create regional replica
                let createRDSReplica = {
                    service: 'RDS',
                    action: 'createDBInstanceReadReplica',
                    region: this.region,
                    parameters: {
                        DBInstanceIdentifier: `replica-${this.region}`,
                        SourceDBInstanceIdentifier: primaryRDSArn,
                        SourceRegion: vpcPeerRegion,
                        KmsKeyId: rdsKms.keyId,
                        PubliclyAccessible: false,
                        DBSubnetGroupName: dbSubnetGroup.dbSubnetGroupName,
                    },
                    physicalResourceId: cr.PhysicalResourceId.of(Date.now().toString())
                }
                let deleteRDSReplica = {
                    service: 'RDS',
                    action: 'deleteDBInstance',
                    region: this.region,
                    parameters: {
                        DBInstanceIdentifier: `replica-${this.region}`,
                        SkipFinalSnapshot: true, // TODO during development, change to retain policy for production
                    },
                    physicalResourceId: cr.PhysicalResourceId.of(Date.now().toString())
                }
                // Create read replica from primary instance
                const rdsReadReplica = new cr.AwsCustomResource(this, `rdsReadReplica`, {
                    onCreate: createRDSReplica,
                    // onUpdate:
                    onDelete: deleteRDSReplica,
                    policy: cr.AwsCustomResourcePolicy.fromSdkCalls({resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE})
                });
```

I have tried looking for alternate methods and failed without having to resort to a full on custom provider to create and wait on isComplete: method and returning a describe instance values

### Proposed Solution
Provide the function to easily create cross region replicas for rds.DatabaseInstanceReadReplica
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseInstanceReadReplica.html

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by indrora** [comment]
Closing as this ticket is stale (over 9 months since last event or relates to the CDKv1). 

If this is still a relevant request in the CDKv2, please open a new ticket.

**Comment by phelian** [comment]
I think this should be implemented
